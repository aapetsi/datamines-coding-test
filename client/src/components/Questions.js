import React, { Component } from "react";

class Questions extends Component {
  state = {
    value: "2"
  };

  onChange = e => {
    this.setState({ value: e.target.value });
  };

  render() {
    const { questions } = this.props;

    const questionOptions = questions.map(question => {
      return (
        <option
          onClick={() => this.props.getQuestionId(this.state.value)}
          value={question.id}
          key={question.id}
        >
          {question.questionDescription}
        </option>
      );
    });

    return (
      <div>
        <select onChange={this.onChange} name="" id="">
          {questionOptions}
        </select>
      </div>
    );
  }
}

export default Questions;
