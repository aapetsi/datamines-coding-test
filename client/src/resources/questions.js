const questions = [
  {
    questionDescription: "How old are you?",
    id: 2,
    responseOptions: [
      {
        option: "13 - 17",
        id: 1
      },
      {
        option: "25-50",
        id: 2
      }
    ]
  },
  {
    questionDescription: "Where do you live?",
    id: 3,
    responseOptions: [
      {
        option: "Accra",
        id: 1
      },
      {
        option: "Tema",
        id: 2
      },
      {
        option: "Other",
        id: 3
      }
    ]
  }
];

export default questions;
