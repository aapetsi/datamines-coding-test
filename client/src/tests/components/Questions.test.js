import React from "react";
import { shallow } from "../../enzyme";
import Questions from "../../components/Questions";
import questions from "../fixtures/questions";

test("should render Questions with questions", () => {
  const wrapper = shallow(<Questions questions={questions} />);
  expect(wrapper).toMatchSnapshot();
});
