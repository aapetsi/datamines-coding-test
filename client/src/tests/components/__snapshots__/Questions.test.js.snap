// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Questions with questions 1`] = `ShallowWrapper {}`;
